src/common.cpp

```diff
@@ -1873,6 +1887,40 @@ std::string get_executable_path(const char *argv0) {
     } else {
         return std::string(buff);
     }
+#elif defined(__OpenBSD__)
+    bool ok = false;
+    std::string narrow;
+    static kvm_t *kd = nullptr;
+    wcstring wargv0 = str2wcstring(argv0);
+    // not sure if I am doing this right??
+    null_environment_t vars = null_environment_t();
```

Comment:
This is an *empty* environment - no variables will be set.

That means `path_get_path` won't look through $PATH.

You need to either get an actual environment, which means passing it from the outside (if you have a parser_t you can use parser_t::vars()), or you need to adjust `path_get_path` so that you can pass a $PATH value in instead of a full environment_t, or you need to redo path_get_path's logic with the normal `getenv("PATH")` $PATH.

I don't think former is possible because this is called really early in startup before the parser is initialized. This would require moving the parser initialization up, and I'm not sure that works before the rest of the init is done.